parser/quakec-parser.js

```diff
@@ -1167,24 +1224,40 @@ Define.definition("$frame",
         Context.token.error("$frame is not a valid type");
     },
     function() {
+        let frameType = null;
+
         while (true) {
             const n = Context.token;
 
-            if (n.arity !== "name") {
+            if (this.range.start.line !== n.range.start.line) {
                 break;
             }
 
-            Context.scope.define(n, this);
-            Parse.advance();
+            if (n.arity !== "name" && n.arity !== "literal") {
+                break;
+            }
 
-            if (Context.token.arity === "literal" && Context.token.type.value === "float") {
-                Parse.advance();
+            if (!frameType) {
+                frameType = n.arity;
+            } else if (frameType !== n.arity) {
+                Context.token.error("Mixed literals and names in frame definitions are invalid.");
+            }
+
+            if (n.arity === "name") {
+                n.value = `$${n.value}`;
+
+                if (Context.scope.def[n.value]) {
+                    delete Context.scope.def[n.value];
+                }
```

Comment:
I'm sure you're gonna raise an eyebrow here. This is because duplicate names are *valid*, the enum value is increased and the last definition of a name ends up being the number being used;

`$frame load1 load1 load1` will generate a constant named `$load1` whose value is 2